### PR TITLE
Numpy v2

### PR DESCRIPTION
**Pull request by scottyhq**
Bring numpy v2 to pangeo-notebook. And nano by popular demand #576  :)

### CODE REVIEW

File: ml-notebook/packages.txt [hunk 1]
```diff
@@ -215,9 +214,7 @@ gcc_impl_linux-64==13.3.0
 gcc_linux-64==13.3.0
 gcm_filters==0.3.0
 gcsfs==2024.9.0
-gdal==3.9.2
```
Comment (scottyhq) [inline ml-notebook/packages.txt:218]: Interesting, gdal command line tools it turns out have been coming from pyogrio, with the latest release it's just  `libgdal-core` dependency https://github.com/geopandas/pyogrio/releases/tag/v0.10.0

Comment (weiji14) [reply ml-notebook/packages.txt:218]: Not just `pyogrio` but also `rasterio` has switched to `libgdal-core`, see https://github.com/conda-forge/pyogrio-feedstock/issues/48 and https://github.com/conda-forge/rasterio-feedstock/issues/303.

File: pangeo-notebook/environment.yml [hunk 2]
```diff
@@ -31,9 +31,11 @@ dependencies:
  - fsspec
  - gcm_filters
  - gcsfs
+ - gdal
  - gh
  - gh-scoped-creds
- - geocube
+ # temporarily drop for numpy>=2 (can use xvec instead)
+ #- geocube
```
Comment (weiji14) [inline pangeo-notebook/environment.yml:38]: Interesting, so geocube pinned to `numpy<2` here - https://github.com/corteva/geocube/pull/175, but the incompatibility should have been fixed in rasterio 1.4 at https://github.com/rasterio/rasterio/pull/3109 if I'm reading correctly. Could open an issue on `geocube` to check, but ok to temporarily remove for now.

**Review by weiji14 — APPROVED** [review]

File: pangeo-notebook/environment.yml (hunk 2 above)
Comment (scottyhq) [reply pangeo-notebook/environment.yml:38]: > but the incompatibility should have been fixed in rasterio 1.4 at https://github.com/rasterio/rasterio/pull/3109 if I'm reading correctly. Could open an issue on geocube to check, but ok to temporarily remove for now. 

That would be great! I took the lazy approach just to see if this was the last remaining hold-out :)

Comment (weiji14) [reply pangeo-notebook/environment.yml:38]: Opened a PR to remove the numpy<2 pin at https://github.com/corteva/geocube/pull/179 :crossed_fingers:

Comment (weiji14) [reply pangeo-notebook/environment.yml:38]: Geocube 0.7.0 is out - https://github.com/conda-forge/geocube-feedstock/pull/41. I'll update the environment.yml file and re-lock

**Review by weiji14 — APPROVED** [review]
Have resolved the merge conflict after merging in #585. @scottyhq, feel free to merge and tag a new version when you're free.

**Pull request merged**